experimental/packages/opentelemetry-instrumentation-fetch/test/fetch.test.ts

```diff
@@ -235,28 +244,22 @@ describe('fetch', () => {
         } else {
           input.text().then(r => (requestBody = r));
         }
-
-        if (init instanceof Request) {
-          // Passing request as 2nd argument causes missing body bug (#2411)
-          response.status = 400;
-          response.statusText = 'Bad Request (Request object as 2nd argument)';
-          reject(new window.Response(JSON.stringify(response), response));
```

Comment:
This branch was replaced entirely. Note that:

1. It is testing for the bug in #2411 which has now be fixed, so this branch is never exercised in practice.
2. To avoid a regression, I replaced the spirit of this with the `assert.ok` at the top.
3. This behavior is misleading and in correct – the real `fetch()` never `reject()` the fetch promise with a response. Even when the server returns an HTTP error (`response.ok === false`), the promise is still _resolved_.

In any case, this branch is not needed anymore.